### PR TITLE
Fix inverted assert

### PR DESCRIPTION
**Pull request by abadams**
This assert in cuda_mat_mul is inverted. err == 0 means success.

### CODE REVIEW

**Comment by shoaibkamil** [comment]
Oh jeez.

**Comment by steven-johnson** [comment]
...no wonder it's been failing so much :-(

**Comment by abadams** [comment]
Seems like it's still failing. On windows it's being done in release mode, so the assert probably wasn't triggering anyway.

**Comment by abadams** [comment]
Looks like the crash is inside cublas :(

**Comment by abadams** [comment]
Now that the asserts are actually on, I'll want to see if any of them trigger.

**Comment by abadams** [comment]
Also, can we globally disable NDEBUG for all our apps and tests? I don't think we ever want it (there's no use of Runtime::Buffer::operator() inside anything we benchmark)

File: apps/cuda_mat_mul/runner.cpp [hunk 1]
```diff
@@ -1,3 +1,8 @@
+// We always want asserts
+#ifdef NDEBUG
+#undef NDEBUG
+#endif
```
Comment (alexreinking) [inline apps/cuda_mat_mul/runner.cpp:4]: BTW - it's always safe to unconditionally `#undef NDEBUG`

**Comment by steven-johnson** [comment]
Do we want to land this with the fprintfs (etc) included?

**Comment by abadams** [comment]
No, it's in a temporary state. I can't tell if it has run yet with the asserts on for real.

**Comment by abadams** [comment]
commit history seems to have gotten messed up.  squashing and force-pushing.

**Comment by steven-johnson** [comment]
Failure is unrelated.

**Comment by steven-johnson** [comment]
(I converted this to 'draft' so it doesn't accidentally get landed yet)

**Comment by abadams** [comment]
At this point this PR was just to help diagnose #5053 

Not worth keeping open as a PR.

**Comment by alexreinking** [comment]
@abadams - do you want to keep the `#undef NDEBUG` in this file, though?

**Comment by abadams** [comment]
I'd rather universally make sure we don't have NDEBUG defined for app testing on the buildbots. I can't guarantee that there aren't asserts elsewhere.

**Pull request closed without merging**